openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -1320,6 +1349,35 @@ def revert_changes(library_key):
         )
     )
 
+    # For each collection, trigger LIBRARY_COLLECTION_UPDATED signal and set background=True to trigger
+    # collection indexing asynchronously.
+    #
+    # This is to update component counts in all library collections,
+    # because there may be components that have been discarded in the revert.
+    for collection in authoring_api.get_collections(learning_package.id):
+        LIBRARY_COLLECTION_UPDATED.send_event(
+            library_collection=LibraryCollectionData(
+                library_key=library_key,
+                collection_key=collection.key,
+                background=True,
+            )
+        )
+
+    # Reindex components that are in collections
+    #
+    # Use case: When a component that was within a collection has been deleted
+    # and the changes are reverted, the component should appear in the
+    # collection again.
+    for component in get_library_components(library_key, in_collections=True):
```

Comment:
```suggestion
    components_in_collections = authoring_api.get_components(
        learning_package.id, draft=True, namespace='xblock.v1',
    ).filter(publishable_entity__collections__isnull=False)

    for component in components_in_collections:
```
I don't think `in_collections` is a useful feature to expose as the public API, so I suggest refactoring like this to avoid it.

Reply:
Updated: https://github.com/openedx/edx-platform/pull/35734/commits/5e771cebf28e066e462fa39e6dac79d713de79f4